src/dbnode/storage/stats/query_stats.go

```diff
@@ -55,33 +57,47 @@ type QueryStatsOptions struct {
 	// MaxDocs limits how many recently queried max
 	// documents are allowed before queries are abandoned.
 	MaxDocs int64
-	// Lookback specifies the lookback period over which stats are aggregated.
-	Lookback time.Duration
+	// MaxDocsLookback specifies the lookback period over which max docs limit is enforced.
+	MaxDocsLookback time.Duration
+	// MaxBytesRead limits how many recently queried bytes
+	// read from disk are allowed before queries are abandoned.
+	MaxBytesRead int64
+	// MaxBytesReadLookback specifies the lookback period over which max bytes read limit is enforced.
+	MaxBytesReadLookback time.Duration
```

Comment:
nit: would it be worth combining these limit/duration tuples into a type? Would make it a little easier to add additional limit types in future

Reply:
I think we're likely to actually remove the other limit if we prove this newer limit is better. So not sure if it'd be worth doing that yet (depends on if we think that limit is still useful, or if we think we'd want more limits later).